src/core/thread/dua_manager.cpp

```diff
@@ -104,11 +105,18 @@ void DuaManager::UpdateDomainUnicastAddress(BackboneRouter::Leader::DomainPrefix
 
 otError DuaManager::GenerateDomainUnicastAddressIid(void)
 {
-    otError error;
+    otError  error;
+    uint16_t dadCounter = mDadCounter;
 
     if ((error = Get<Utils::Slaac>().GenerateIid(mDomainUnicastAddress, NULL, 0, &mDadCounter,
                                                  false /* not allow random IID if no valid IID */)) == OT_ERROR_NONE)
     {
+        if (dadCounter != mDadCounter)
```

Comment:
Why change `mDadCounter` back to `dadCounter`?
Should pass `&dadCounter` to `Get<Utils::Slaac>().GenerateIid` instead of `&mDadCounter`?

Reply:
oh, yes, careless mistake. will update.